src/readers/morphologyHDF5.cpp

```diff
@@ -318,13 +318,13 @@ void MorphologyHDF5::_read(const std::string& groupName,
                            const std::string& datasetName,
                            unsigned int expectedDimension,
                            T& data) {
-    if (!_group.exist(groupName)) {
+    if (groupName != "" && !_group.exist(groupName)) {
         throw(
             RawDataError("Reading morphology '" + _uri + "': Missing required group " + groupName));
     }
-    const auto group = _group.getGroup(groupName);
+    const auto group = groupName == "" ? _group : _group.getGroup(groupName);
 
-    if (!_group.exist(groupName)) {
+    if (!group.exist(datasetName)) {
```

Comment:
is this the bug you reference in the PR description?

Reply:
Yes, when adding a test with a container that contained morphologies with perimeters, it would complain that there was to dataset called `"perimeters"`.

What makes me slightly wary is that we've used containers in TouchDetector. However, it's not impossible, that we've simply never used more complex morphologies with a perimeter.

The proposed change I believe is correct, whether the previous was a bug I'm not 100% sure. I strongly suspect it was, but as mentioned it would have been quite good at hiding.

Reply:
Sorry! I've not reread my description. Yes, the `datasetName` is relative to `groupName`. For fancy morphologies the `groupName` might be something like `"organelles/mitochondria"` and the `datasetName` would be `"points"`.

Hence, if we want to check that the dataset exists, we need to check the `group` (not `_group`) and check `datasetName`. It's highly likely it's a copy-pasting bug from the check on line 321.